### PR TITLE
Fix/use full hash of shard

### PR DESCRIPTION
**Pull request by oleksiylukin**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardInfoDownloader.java [hunk 1]
```diff
@@ -314,13 +381,20 @@ public double getShardWeight(Long shardId) {
...
         }
+        //we can use environment variable to fore download of some shard
+        Long predefined = readManuallyDefinedShardId();
+        if(predefined!=null){
+            boolean exists = res.get(predefined)!=null;
+            if(exists){
+                res.replace(predefined, 20.0D);
+                log.info("Downloading of shard {} is forced by enviromnent variable",predefined);
+            }else{
+                log.info("Shard ID {} predefined in environment variable is not usable or does not exist");
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/files/shards/ShardInfoDownloader.java]: not needed {} in message

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java [hunk 2]
```diff
@@ -78,21 +78,26 @@ private void recalculatePrunableArchiveHashes() {
...
                     shard.setPrunableZipHash(null);
                     shardDao.updateShard(shard);
                     Files.deleteIfExists(prunableArchivePath);
+                    ops = new ChunkedFileOps("");
                 } else {
                     String zipName = "shard-" + shard.getShardId() + ".zip";
-                    ChunkedFileOps ops = zip.compressAndHash(tempDirectory.resolve(zipName).toAbsolutePath().toString(), tempDirectory.toAbsolutePath().toString(), 0L, null, false);
-                    fileChangedEvent.select(new AnnotationLiteral<FileChangedEvent>(){}).fireAsync(ops);
-                    log.debug("Firing 'FILE_CHANDED' event {}", ops.getFileId());
+                    ops = zip.compressAndHash(tempDirectory.resolve(zipName).toAbsolutePath().toString(), tempDirectory.toAbsolutePath().toString(), 0L, null, false);
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java]: Does variable  `String zipName = "shard-" + shard.getShardId() + ".zip";`
get correct value here ??
How that will process : `zip.compressAndHash(tempDirectory.resolve(zipName)` with value like = 'shard-1.zip' ??

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java [hunk 3]
```diff
@@ -78,21 +78,26 @@ private void recalculatePrunableArchiveHashes() {
...
+                    if( ops==null || ! ops.isHashedOK()){
+                        log.error("Can not zip file: {}", zipName);
+                    }
                     byte[] hash = ops.getFileHash();
-                    ops.setFileId(shardNameHelper.getFullShardId(shard.getShardId(), chainId));
+                    ops.setFileId(shardNameHelper.getFullShardPrunId(shard.getShardId(), chainId));
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java]: The name is correct here.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java (hunk 2 above)
Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java]: this is just temp file

Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardPrunableZipHashCalculator.java]: May be better to give it name like 'shard-temp-....' ?

**Pull request merged**